bundles/org.openhab.binding.ecovacs/src/main/resources/OH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,351 @@
+		<state pattern="%d %unit%" readOnly="true"></state>
+	</channel-type>
+
+	<channel-type id="last-clean-mode">
+		<item-type>String</item-type>
+		<label>Last Cleaning Mode</label>
+		<description>Operation mode used in last completed cleaning run</description>
+		<state readOnly="true">
+			<options>
+				<option value="auto">@text/ecovacs.cleaning-mode.auto</option>
+				<option value="edge">@text/ecovacs.cleaning-mode.edge</option>
+				<option value="spot">@text/ecovacs.cleaning-mode.spot</option>
+				<option value="spotArea">@text/ecovacs.cleaning-mode.spotArea</option>
+				<option value="customArea">@text/ecovacs.cleaning-mode.customArea</option>
+				<option value="singleRoom">@text/ecovacs.cleaning-mode.singleRoom</option>
+			</options>
+		</state>
+	</channel-type>
+
+	<channel-type id="last-clean-map">
+		<item-type>Image</item-type>
+		<label>Last Clean Map</label>
+		<description>Cleaning map for last completed cleaning run</description>
+		<state readOnly="true"/>
+	</channel-type>
+
+	<channel-type id="auto-empty" advanced="true">
+		<item-type>Switch</item-type>
+		<label>Auto Empty</label>
+		<description>Automatically empty dust bin in station</description>
+	</channel-type>
+
+	<channel-type id="cleaning-passes" advanced="true">
+		<item-type>Number</item-type>
+		<label>Cleaning Passes</label>
+		<description>Number of cleaning passes used by default (if not overridden in command)</description>
+		<state min="1" max="2" step="1" pattern="%d"/>
+	</channel-type>
+
+	<channel-type id="continuous-cleaning" advanced="true">
+		<item-type>Switch</item-type>
+		<label>Continuous Cleaning</label>
+		<description>Automatically resume unfinished cleaning after charging</description>
+	</channel-type>
+
+	<channel-type id="suction-power">
+		<item-type>String</item-type>
+		<label>Cleaning Power Level</label>
+		<state>
+			<options>
+				<option value="silent">Silent</option>
+				<option value="normal">Normal</option>
+				<option value="high">Maximum</option>
+				<option value="higher">Maximum+</option>
+			</options>
+		</state>
+	</channel-type>
+
+	<channel-type id="true-detect-3d" advanced="true">
+		<item-type>Switch</item-type>
+		<label>True Detect 3D</label>
+	</channel-type>
+
+	<channel-type id="water-system-present">
+		<item-type>Switch</item-type>
+		<label>Water System Present</label>
+		<description>Water plate with mop attached to device?</description>
+		<state readOnly="true"/>
+	</channel-type>
+
+	<channel-type id="water-amount" advanced="true">
+		<item-type>String</item-type>
+		<label>Mopping Water Amount</label>
+		<state>
+			<options>
+				<option value="low">Low</option>
+				<option value="medium">Medium</option>
+				<option value="high">High</option>
+				<option value="veryhigh">Very high</option>
+			</options>
+		</state>
+	</channel-type>
+
+	<channel-type id="wifi-rssi" advanced="true">
+		<item-type>Number:Power</item-type>
+		<label>Wi-Fi Signal Strength</label>
```

Comment:
Description missing. Consider also:
```suggestion
		<label>Wi-Fi Signal Strength</label>
		<category>QualityOfService</category>
```

Reply:
:+1: for category, but is a description mandatory? Thinking of how a description could be worded that is not just a copy of the label.